tests/integration/test_rest_schema.py

```diff
@@ -1730,19 +1730,17 @@ def test_move_nested_field_after_first(catalog: Catalog) -> None:
     with tbl.update_schema() as schema_update:
         schema_update.move_before("struct.data", "struct.count")
```

Comment:
this makes me think that the Field ordering does matter...

@Fokko wdyt

Reply:
First of all, thanks for digging into this 🎉 

Technically the ordering does not matter when you write the data, because when reading we're correcting the order using this one: https://github.com/apache/iceberg-python/blob/d02d7a1a7fefe4c2b68aef68c8029a2d5fc2b253/pyiceberg/io/pyarrow.py#L1143

Maybe we should also use that visitor when writing (instead of the PyArrow cast) introduced in https://github.com/apache/iceberg-python/pull/523

Reply:
make sense, thanks!

Reply:
> Maybe we should also use that visitor when writing (instead of the PyArrow cast) introduced

We're relying on pyarrow `cast` to translate some pyarrow data types into corresponding Iceberg-supported data types. Such as `large_string` -> `string`. Since `LargeString` is not an Iceberg-supported data type, we cannot use `to_requested_schema`. Maybe it's possible to cast pyarrow `LargeString` into Iceberg `String`.

Reply:
I don't think we want to do that. The `String` and `LargeString` are an Arrow encoding detail (similar to the categorial type). Maybe we should have a different version of `to_requested_schema` where we don't cast, and just keep the original types?